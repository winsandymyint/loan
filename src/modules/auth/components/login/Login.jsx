import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Field } from 'redux-form'
import { Button, Col, Divider, Icon, Row, ReduxFormInput } from 'components'
import { email, required } from 'utils/formValidation'
import { translate } from 'utils/translation'
import storage from 'utils/storage'
import logoFull from 'assets/images/logo-full.png'

class Login extends Component {
    componentWillMount = () => {
        const { history } = this.props

        if (storage.isAuthenticated()) {
            history.push('/')
        }
    }

    handleForgotPassword = () => {
        this.props.history.push('forgot-password')
    }

    handleFormSubmit = ({ email, password }) => {
        this.props.login(email, password)
    }

    handleGoogleFormSubmit = event => {
        event.preventDefault()
        this.props.loginWithGoogle()
    }

    render() {
        const { handleSubmit, isLoadingLogin } = this.props

        return (
            <Row align='middle' className='login' justify='center' type='flex'>
                <Col className='login__form' span={6}>
                    <form onSubmit={handleSubmit(this.handleFormSubmit)}>
                        <div className='login__form-image'>
                            <img src={logoFull} />
                        </div>
                        <Field
                            component={ReduxFormInput}
                            name='email'
                            props={{
                                autoComplete: 'email',
                                disabled: isLoadingLogin,
                                prefix: <Icon type='mail' />,
                                placeholder: translate('INPUT_EMAIL_PLACEHOLDER'),
                                type: 'email'
                            }}
                            validate={[required, email]}
                        />
                        <Field
                            component={ReduxFormInput}
                            name='password'
                            props={{
                                autoComplete: 'current-password',
                                prefix: <Icon type='key' />,
                                placeholder: translate('INPUT_PASSWORD_PLACEHOLDER'),
                                type: 'password'
                            }}
                            validate={required}
                        />
                        <div className='login__form-forgot-password'>
                            <span onClick={this.handleForgotPassword}>{translate('AUTH_FORGOT_PASSWORD_TEXT')}</span>
                        </div>
                        <div className='login__form-login-btn'>
                            <Button
                                htmlType='submit'
                                icon='login'
                                loading={isLoadingLogin}
                                name='login-btn'
                                type='primary'
                                value={translate('AUTH_LOGIN_TEXT')}
                            />
                        </div>
                    </form>
                    <Divider />
                    <div>
                        <Row className='login__form-social-login-btn' justify='space-around' type='flex'>
                            <Button
                                icon='google'
                                loading={isLoadingLogin}
                                name='login-btn'
                                onClick={this.handleGoogleFormSubmit}
                                value={translate('AUTH_LOGIN_GOOGLE_TEXT')}
                            />
                        </Row>
                    </div>
                </Col>
            </Row>
        )
    }
}

Login.propTypes = {
    handleSubmit: PropTypes.func,
    history: PropTypes.object,
    isLoadingLogin: PropTypes.bool,
    login: PropTypes.func,
    loginWithGoogle: PropTypes.func,
    submitting: PropTypes.bool
}
export default Login
