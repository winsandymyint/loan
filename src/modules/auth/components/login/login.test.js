/* global test expect shallow */

import React from 'react'
import Login from './Login'

test('component matches snapshot', () => {
    const wrapper = shallow(<Login handleSubmit={() => undefined} />)
    expect(wrapper).toMatchSnapshot()
})

test('render props not in root node', () => {
    const wrapper = shallow(<Login handleSubmit={() => undefined} isLoading />)
    expect(wrapper.instance().props).toHaveProperty('isLoading', true)
})
