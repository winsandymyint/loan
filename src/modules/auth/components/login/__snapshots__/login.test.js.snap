// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component matches snapshot 1`] = `
<Row
  align="middle"
  className="login"
  gutter={0}
  justify="center"
  type="flex"
>
  <Col
    className="login__form"
    span={6}
  >
    <form>
      <div
        className="login__form-image"
      >
        <img
          src="test-file-stub"
        />
      </div>
      <Field
        component={[Function]}
        name="email"
        props={
          Object {
            "autoComplete": "email",
            "disabled": undefined,
            "placeholder": "Enter your email",
            "prefix": <Icon
              spin={false}
              theme="outlined"
              type="mail"
            />,
            "type": "email",
          }
        }
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
      />
      <Field
        component={[Function]}
        name="password"
        props={
          Object {
            "autoComplete": "current-password",
            "placeholder": "Enter your password",
            "prefix": <Icon
              spin={false}
              theme="outlined"
              type="key"
            />,
            "type": "password",
          }
        }
        validate={[Function]}
      />
      <div
        className="login__form-forgot-password"
      >
        <span
          onClick={[Function]}
        >
          forgot password ?
        </span>
      </div>
      <div
        className="login__form-login-btn"
      >
        <Button
          disabled={false}
          htmlType="submit"
          icon="login"
          loading={false}
          name="login-btn"
          size="default"
          type="primary"
          value="Login"
        />
      </div>
    </form>
    <Divider
      dashed={false}
      type="horizontal"
    />
    <div>
      <Row
        align="top"
        className="login__form-social-login-btn"
        gutter={0}
        justify="space-around"
        type="flex"
      >
        <Button
          disabled={false}
          htmlType="button"
          icon="google"
          loading={false}
          name="login-btn"
          onClick={[Function]}
          size="default"
          type="default"
          value="Login with Google"
        />
      </Row>
    </div>
  </Col>
</Row>
`;
